gradio/deploy_space.py

```diff
@@ -0,0 +1,183 @@
+    if formatted_title != title:
+        print(f"Formatted to {formatted_title}. ")
+    configuration["title"] = formatted_title
+
+    app_file = None
+    for file in os.listdir(repo_directory):
+        file_path = os.path.join(repo_directory, file)
+        if (not os.path.isfile(file_path)) or (
+            not file.endswith(".py") and not file.endswith(".ipynb")
+        ):
+            continue
+
+        with open(file_path, encoding="utf-8", errors="ignore") as f:
+            content = f.read()
+            if "import gradio" in content:
+                app_file = file
+                break
+
+    app_file = (
+        input(f"Enter Gradio app file {f'[{app_file}]' if app_file else ''}: ")
+        or app_file
+    )
+    if not app_file or not os.path.exists(app_file):
+        raise FileNotFoundError("Failed to find Gradio app file.")
+    if app_file.endswith(".ipynb"):
+        configuration["notebook_file"] = app_file
+        app_file = "_" + configuration["notebook_file"].replace(".ipynb", ".py")
+    configuration["app_file"] = app_file
+
+    emoji_set = "🤯🤖🧠🐶👑💥🐮🎁🐙🦋"
+    default_emoji = random.choice(emoji_set)
+    configuration["emoji"] = (
+        input(f"Enter Spaces Card emoji [{default_emoji}]: ") or default_emoji
+    )
+
+    color_set = ["red", "yellow", "green", "blue", "indigo", "purple", "pink", "gray"]
+    default_color = random.choice(color_set)
+    color = input(f"Enter Spaces Card color [{default_color}]: ") or default_color
+    configuration["colorFrom"] = color
+    configuration["colorTo"] = color
+    configuration["sdk"] = "gradio"
+    huggingface_hub.metadata_save(readme_file, configuration)
+
+    configuration["hardware"] = (
+        input(
+            f"Enter Spaces hardware ({', '.join(hardware.value for hardware in huggingface_hub.SpaceHardware)}) [cpu-basic]: "
+        )
+        or "cpu-basic"
+    )
+    secrets = {}
+    if input("Any Spaces secrets (y/n) [n]: ") == "y":
+        while True:
+            secret_name = input("Enter secret name (leave blank to end): ")
+            if not secret_name:
+                break
+            secret_value = input(f"Enter secret value for {secret_name}: ")
+            secrets[secret_name] = secret_value
+    configuration["secrets"] = secrets
+
+    requirements_file = os.path.join(repo_directory, "requirements.txt")
+    if (
+        not os.path.exists(requirements_file)
+        and input("Create requirements.txt file? (y/n) [n]: ").lower() == "y"
+    ):
+        input(
+            "Created requirements.txt file. Please add any dependencies to this file now. Press enter to continue. "
+        )
+
+    return configuration
+
+
+def format_title(title: str):
+    title = title.replace(" ", "_")
+    title = re.sub(r"[^a-zA-Z0-9\-._]", "", title)
+    title = re.sub("-+", "-", title)
+    while title.startswith("."):
+        title = title[1:]
+    return title
+
+
+def notebook_to_script(notebook_path, script_path):
```

Comment:
This conversion script messed up for me: https://huggingface.co/spaces/freddyaboulton/calculator-notebook-deploy/tree/main

I wonder if we can use an existing tool for this (optional dependency) or leave notebook deploy out of scope for now.

Reply:
I think you're right. Would be nice to support a "push from colab" but I think it's out of scope for now.

Reply:
So just to confirm, we don't support automagically pushing from a notebook? It'd be a really slick workflow but we can leave it as a future PR.

Reply:
And if we don't support conversion of notebooks, I don't think having a `gr.deploy()` makes much sense. You'd always do `gradio deploy` from the terminal